### PR TITLE
new audio engine

### PR DESCRIPTION
**Pull request by toddmedema**

### CODE REVIEW

File: app/actions/Card.tsx [hunk 1]
```diff
@@ -25,8 +26,10 @@ export const toCard = remoteify(function toCard(a: ToCardArgs, dispatch?: Redux.
   if (!a.noHistory) {
     dispatch({type: 'PUSH_HISTORY'});
   }
+  if (state.audio.intensity > 0 && (a.name !== 'QUEST_CARD' || a.phase === null || a.phase === 'ROLEPLAY')) {
```
Comment (smartin015) [inline app/actions/Card.tsx]: Why is intensity set in the card? Could we instead do this in the Combat or Audio JSX element?

File: app/cardtemplates/combat/Actions.tsx [hunk 2]
```diff
@@ -59,6 +61,10 @@ export const initCustomCombat = remoteify(function initCustomCombat(a: InitCusto
   return {};
 });
 
+function getAudioIntensity(currentTier: number, maxTier: number, deadAdventurers: number, roundCount: number): number {
+  return Math.round(Math.min(36, 2 * currentTier + 1 * maxTier + 4 * deadAdventurers + 0.5 * roundCount));
```
Comment (smartin015) [inline app/cardtemplates/combat/Actions.tsx]: Pls add comment describing magic numbers (or better, constant values)

File: app/cardtemplates/combat/Actions.tsx [hunk 3]
```diff
@@ -59,6 +61,10 @@ export const initCustomCombat = remoteify(function initCustomCombat(a: InitCusto
   return {};
 });
 
+function getAudioIntensity(currentTier: number, maxTier: number, deadAdventurers: number, roundCount: number): number {
```
Comment (smartin015) [inline app/cardtemplates/combat/Actions.tsx]: What about `calculateAudioIntensity`? This doesn't get an existing value, but rather makes one.

File: app/actions/Card.tsx (hunk 1 above)
Comment (toddmedema) [reply app/actions/Card.tsx]: So, my line of thinking was that there are a lot of ways to get out of combat, and if we fail at catching even one of them, their app will be stuck playing music forever. So an over-aggressive, global catch seemed like a pretty good solution from a user experience perspective. Open to other ways we could achieve that :) Perhaps by passing node information to the audio element, and having it do this check?

File: app/cardtemplates/combat/Actions.tsx (hunk 3 above)
Comment (toddmedema) [reply app/cardtemplates/combat/Actions.tsx]: 👍

File: app/cardtemplates/combat/Actions.tsx (hunk 2 above)
Comment (toddmedema) [reply app/cardtemplates/combat/Actions.tsx]: 👍

File: app/actions/Card.tsx (hunk 1 above)
Comment (toddmedema) [reply app/actions/Card.tsx]: (fixed - moved to audio element)

**Pull request merged**